src/components/NcModal/NcModal.vue

```diff
@@ -420,13 +420,24 @@ export default {
 				return ['small', 'normal', 'large', 'full'].includes(size)
 			},
 		},
+
 		/**
 		 * Declare if the modal can be closed
 		 */
 		canClose: {
 			type: Boolean,
 			default: true,
 		},
+
+		/**
+		 * Close the modal if the user clicked outside of the modal
+		 * Only relevant if `canClose` is set to true.
+		 */
+		closeOnClickOutside: {
+			type: Boolean,
+			default: true,
+		},
```

Comment:
In a standard form of passing props Vue enforces HTML standard aka:

```html
<input /> <!-- enabled-->
<input disabled /> <!-- disabled -->
```

But boolean props with default `true` requires to always use longhand form over HTML Standard:

```vue
<NcModal /> <!-- closes on click -->
<NcModal close-on-click-outside /> <!-- also closes on click, passing bool prop does nothing -->
<NcModal :close-on-click-outside="false" /> <!-- changes behavior, doesn't close on click -->
<NcModal close-on-click-outside="false" /> <!-- casts string "false" to true and closes on click -->
```

because of that it is usually recommended to make bool props with `false` default value:

```vue
<!-- With default false boolean prop -->
<NcModal /> <!-- closes on click -->
<NcModal ignore-click-outside /> <!-- changes behavior, doesn't close on click -->
```

- https://eslint.vuejs.org/rules/prefer-true-attribute-shorthand.html
- https://eslint.vuejs.org/rules/no-boolean-default.html

Reply:
Thank you :)
I know but did not wanted to make this a breaking change by changing the default behavior and a negated property seems to be confusing.
Maybe we can risk another breaking change?